### PR TITLE
chore(node-core): decouple runtime and app module

### PR DESCRIPTION
**Pull request by archbear**
## Summary by CodeRabbit

- **New Features**
  - Introduced an `ABCIMiddleware` struct that enhances interaction between ABCI and Beacon logic.
  
- **Enhancements**
  - Updated middleware functions and types to improve flexibility and functionality in managing Beacon blocks and states.
  - Added new methods to the `BeaconState` interface for better access to beacon state information.
  - Updated various middleware components to use more specific type parameters and structures.

- **Refactor**
  - Restructured import paths and type declarations to streamline middleware and runtime components.
  
- **Chores**
  - Removed outdated imports and adjusted dependencies in several components for cleaner integration and better maintainability.

### CODE REVIEW

**Review by archbear — COMMENTED** [review]
stat padding yippee

**Review by ocnc2 — APPROVED** [review]
lgtm

**Comment by ocnc2** [comment]
plan:
1. make storage backend its own cosmos AppModule at long last
2. remove the wrapped BKRuntime module
3. expose storage backend keeper to our BKRuntime

**Comment by itsdevbear** [comment]
wait is ur plan to have two modules then? or have the app call our runtime for Finalize Block ns tuff?

File: mod/node-core/pkg/components/module/depinject.go [hunk 1]
```diff
@@ -34,16 +35,17 @@ import (
...
+			ProvideModule, // for the github contribution lolololololol
 		),
 	)
 }
 
 // ModuleInput is the input for the dep inject framework.
 type ModuleInput struct {
 	depinject.In
-	Runtime *components.BeaconKitRuntime
+	storageBackend              components.StorageBackend
```
Comment (ocnc2) [inline mod/node-core/pkg/components/module/depinject.go:46]: this needs to be capitalized for depinject to resolve

File: mod/node-core/pkg/components/storage.go [hunk 2]
```diff
@@ -23,29 +23,18 @@ package components
...
-	*dastore.Store[*types.BeaconBlockBody],
-	*types.BeaconBlockBody,
-	BeaconState,
-	*datypes.BlobSidecars,
-	*types.Deposit,
-	*depositdb.KVStore[*types.Deposit],
-]
-
 // StorageBackendInput is the input for the ProvideStorageBackend function.
 type StorageBackendInput struct {
```
Comment (archbear) [inline mod/node-core/pkg/components/storage.go:39]: remove

File: mod/runtime/pkg/runtime/mod.go
Comment (archbear) [inline mod/runtime/pkg/runtime/mod.go:1]: tbh the more i think about it we should have a "types.go" or something in components which is a list of type aliases with the hard types injected there, this shouldn't be here cause runtime should be an isolated module

File: mod/node-core/pkg/builder/extended_options.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+import (
+	"github.com/berachain/beacon-kit/mod/primitives"
+	"github.com/berachain/beacon-kit/mod/runtime/pkg/comet"
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Options that extend our default baseapp options to be called by cosmos
+// when buildinig the app.
```
Comment (archbear) [inline mod/node-core/pkg/builder/extended_options.go:31]: typo

File: mod/node-core/pkg/components/storage.go (hunk 2 above)
Comment (ocnc2) [reply mod/node-core/pkg/components/storage.go:38]: what u want removed?

File: mod/runtime/pkg/middleware/validator.go [hunk 4]
```diff
@@ -52,13 +52,7 @@ type ValidatorMiddleware[
...
 	},
 	BeaconBlockBodyT types.RawBeaconBlockBody,
-	BeaconStateT interface {
-		ValidatorIndexByPubkey(pk crypto.BLSPubkey) (math.ValidatorIndex, error)
-		GetBlockRootAtIndex(slot uint64) (primitives.Root, error)
-		ValidatorIndexByCometBFTAddress(
-			cometBFTAddress []byte,
-		) (math.ValidatorIndex, error)
-	},
+	BeaconStateT BeaconState,
```
[the remark this answers is not in the document]
Comment (archbear) [reply mod/runtime/pkg/middleware/validator.go:55]: what is bro yappin about

[the remark this answers is not in the document]
Comment (archbear) [reply mod/runtime/pkg/middleware/validator.go:55]: sorry sir 🫡

**Pull request merged**